### DISCUSSION TITLE
TypeError: Cannot read properties of undefined (reading 'toJSON')

### DISCUSSION THREAD

**Discussion by arnon001** [discordjs/discord.js#8419] [Q&A] [answered]
### Which package is this bug report for?

discord.js

### Issue description

I'm using the Discord.js guide for slash commands, and i'm getting this error:
![image](https://user-images.githubusercontent.com/58848496/182573514-3d20f57d-62d2-4a44-972e-edf109107e6d.png)
I'm copying all just like in here: https://discordjs.guide/interactions/slash-commands.html#guild-commands

### Code sample

```typescript
const { REST } = require('@discordjs/rest');
const { Routes, Client, GatewayIntentBits } = require('discord.js');
const { token } = require('./config.json');
const fs = require('node:fs');
const client = new Client({ intents: [GatewayIntentBits.Guilds] });

const commands = [];
const commandFiles = fs.readdirSync('./commands').filter(file => file.endsWith('.js'));

// Place your client and guild ids here
const clientId = 'myclientidisprivatern';

for (const file of commandFiles) {
	const command = require(`./commands/${file}`);
	commands.push(command.data.toJSON());
}

const rest = new REST({ version: '10' }).setToken(token);

(async () => {
	try {
		console.log('Started refreshing application (/) commands.');

		await rest.put(
			Routes.applicationGuildCommands(clientId),
			{ body: commands },
		);

		console.log('Successfully reloaded application (/) commands.');
	} catch (error) {
		console.error(error);
	}
})();
client.login(token)
```

### Package version

14.1.2

### Node.js version

18.7.0

### Operating system

Windows

### Priority this issue should have

Medium (should be fixed soon)

### Which partials do you have configured?

No Partials

### Which gateway intents are you subscribing to?

No Intents

**Comment by SuperchupuDev** [comment, marked as answer]
One of your commands doesn't have a `data` property exported. You should make sure no command files are empty and export the command builder as `data`

**Reply by Flare-0** [reply]
hey thanks for the answer u r a lifesaver
